pkg/basicserver/basic_server.go

```diff
@@ -39,8 +33,8 @@ type Server interface {
 	Run() error
 	// Close closes the server.
 	Close()
-	// GetPrimary returns the primary of the server.
-	GetPrimary() MemberProvider
+	// GetLeaderListenUrls gets service endpoints from the leader in election group.
+	GetLeaderListenUrls() []string
```

Comment:
how about return `string` directly? There is only one leader or none.

Reply:
It's the listenUrls passed to the pd which becomes leader. It can be multiple urls and there is one place in admin actually accessing all urls.

Reply:
I found we use listenUrls[0] as primary by default.

Reply:
Yes, so any problem with that? Using []string is also for keeping compatibility with both pdpb.Member and tsopb.Participant. For former, it's clientUrls which is an array. For the latter, it's the listenUrl, a.k.a, listenUrls[0], and we may extend it to multiple addresses of participants which join the election group and use listenUrls[0] as the primary.